src/game/enemy/systems.rs

```diff
@@ -82,74 +121,132 @@ fn spawn_health_bar(builder: &mut ChildBuilder) {
             },
-            transform: Transform::from_translation(Vec3::new(0., 25., 3.)),
+            transform: Transform::from_translation(Vec3::new(0., 70., 3.)),
             ..default()
         },
-        HealthBar
+        HealthBar,
     ));
 }
 
 pub fn despawn_enemies(mut commands: Commands, enemy_q: Query<Entity, With<Enemy>>) {
     println!("Removing all enemies");
     enemy_q.iter().for_each(|enemy_entity| {
-        commands.entity(enemy_entity).despawn();
+        commands.entity(enemy_entity).despawn_recursive();
     });
 }
 
-// This will change a lot once
-pub fn spawn_enemies(
+pub fn spawn_initial_enemies(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut word_bank: ResMut<WordBank>,
+    word_q: Query<&Word, (With<Word>, Without<InputText>)>,
+    enemy_spawn: Res<EnemySpawns>,
+    enemy_handles: Res<EnemyHandles>,
+) {
+    println!("Spawning initial enemies");
+    let font = asset_server.load("fonts/fyodor/truetype/Fyodor-BoldCondensed.ttf");
+    let mut rng = thread_rng();
+    for pos in enemy_spawn.enemies.as_slice() {
+        let animation_indices = EnemyAnimations::SovietIdle.get_indices();
+        commands
+            .spawn((
+                SpriteSheetBundle {
+                    texture_atlas: enemy_handles.soviet_idle.clone(),
+                    sprite: TextureAtlasSprite::new(
+                        rng.gen_range(animation_indices.0..animation_indices.1),
+                    ),
+                    transform: Transform::from_xyz(pos.x, pos.y, 1.),
+                    ..default()
+                },
+                animation_indices,
+                EnemyAnimations::SovietIdle.get_timer(),
+                AnimateSprite,
+                Enemy::default(),
+            ))
+            .with_children(|builder| {
+                spawn_health_bar(builder);
+                spawn_word(
+                    builder,
+                    word_bank.get_word(WordComplexity::Medium, &word_q).as_str(),
+                    &font,
+                );
+            });
+    }
+}
+
+/// Set resources once we progress to a new level
+pub fn init_enemy_level_info(
+    mut commands: Commands,
+    level_info_q: Query<&LevelInfo>,
+    mut level_complete_event_reader: EventReader<LevelCompletedEvent>,
+) {
+    for level in level_complete_event_reader.iter() {
+        if let Some(level_info) = level_info_q.iter().nth(level.0) {
+            println!("Inserting resources");
+            commands.insert_resource(EnemySpawnTimer {
+                timer: Timer::from_seconds(level_info.spawn_rate, TimerMode::Repeating),
+            });
+            commands.insert_resource(EnemySpawns {
+                enemies: level_info.enemies.clone(),
+            });
+        }
+    }
+}
+
+/// Tick spawn timer while in game and unpaused
+pub fn tick_enemy_spawn_timer(mut enemy_spawn_timer: ResMut<EnemySpawnTimer>, time: Res<Time>) {
+    enemy_spawn_timer.timer.tick(time.delta());
```

Comment:
possible i misunderstand how the timer works, but feels like we not only could but would want to have this in the spawn_enemies_gradually method, since the timer.finished() would consume it "finishing" wouldnt it?

Reply:
It doesn't matter much I don't think. I'll change it, but I originally kept it separate because of too many arguments